src/fsharp/NameResolution.fs

```diff
@@ -3025,8 +3025,7 @@ let InfosForTyconConstructors (ncenv:NameResolver) m ad (tcref:TyconRef) =
             | Item.CtorGroup(nm,ctorInfos) -> 
                 let ctors = 
                     ctorInfos 
-                    |> List.filter (IsMethInfoAccessible amap m ad)
-                    |> List.filter (MethInfoIsUnseen g m typ >> not)
+                    |> List.filter (fun x -> IsMethInfoAccessible amap m ad x && (MethInfoIsUnseen g m typ x |> not))
```

Comment:
In the compiler code, please don't pipe into ``not`` . At least if it's the only pipe in a chain.

So here just use ``not (MethInfoIsUnseen g m typ)``.